nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/data/explorer/StandardKustoIngestService.java

```diff
@@ -0,0 +1,362 @@
+            .description("Authentication method for access to Azure Data Explorer")
+            .required(true)
+            .defaultValue(KustoAuthenticationStrategy.MANAGED_IDENTITY.getValue())
+            .allowableValues(KustoAuthenticationStrategy.class)
+            .build();
+
+    public static final PropertyDescriptor APPLICATION_CLIENT_ID = new PropertyDescriptor
+            .Builder().name("Application Client ID")
+            .displayName("Application Client ID")
+            .description("Azure Data Explorer Application Client Identifier for Authentication")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor APPLICATION_KEY = new PropertyDescriptor
+            .Builder().name("Application Key")
+            .displayName("Application Key")
+            .description("Azure Data Explorer Application Key for Authentication")
+            .required(true)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(AUTHENTICATION_STRATEGY, KustoAuthenticationStrategy.APPLICATION_CREDENTIALS.getValue())
+            .build();
+
+    public static final PropertyDescriptor APPLICATION_TENANT_ID = new PropertyDescriptor.Builder()
+            .name("Application Tenant ID")
+            .displayName("Application Tenant ID")
+            .description("Azure Data Explorer Application Tenant Identifier for Authentication")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(AUTHENTICATION_STRATEGY, KustoAuthenticationStrategy.APPLICATION_CREDENTIALS.getValue())
+            .build();
+
+    public static final PropertyDescriptor CLUSTER_URI = new PropertyDescriptor
+            .Builder().name("Cluster URI")
+            .displayName("Cluster URI")
+            .description("Azure Data Explorer Cluster URI")
+            .required(true)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            AUTHENTICATION_STRATEGY,
+            APPLICATION_CLIENT_ID,
+            APPLICATION_KEY,
+            APPLICATION_TENANT_ID,
+            CLUSTER_URI));
+
+    public static final Pair<String, String> NIFI_SINK = Pair.of("processor", "nifi-sink");
+
+    private volatile QueuedIngestClient queuedIngestClient;
+
+    private volatile ManagedStreamingIngestClient managedStreamingIngestClient;
+
+    private volatile Client executionClient;
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    /**
+     * @param context the configuration context
+     * @throws InitializationException if unable to create a database connection
```

Comment:
This `throws` line is not correct. It is probably better to remove the Javadoc comment if it is not completely written.